lambdas/nino-check/src/hmrc-apis/pdv.ts

```diff
@@ -0,0 +1,67 @@
+import { PdvApiErrorBody, PdvApiInput, PdvApiResponseBody, PdvConfig, PdvFunctionOutput } from "./types/pdv";
+import { logger } from "../../../common/src/util/logger";
+import { captureLatency } from "../../../common/src/util/metrics";
+import { safeStringifyError } from "../../../common/src/util/stringify-error";
+
+export async function matchUserDetailsWithPdv(
+  { apiUrl, userAgent }: PdvConfig,
+  oAuthToken: string,
+  apiInput: PdvApiInput
+): Promise<PdvFunctionOutput> {
+  const [response, latency] = await captureLatency("MatchingHandler", () =>
+    fetch(apiUrl, {
+      method: "POST",
+      headers: {
+        "Content-Type": "application/json",
+        "User-Agent": userAgent,
+        Authorization: `Bearer ${oAuthToken}`,
+      },
+      body: JSON.stringify(apiInput),
+    })
+  );
+
+  logger.info({
+    message: "PDV API response received",
+    url: apiUrl,
+    status: response.status,
+    latencyInMs: latency,
+  });
+
+  const txn = response.headers.get("x-amz-cf-id") ?? "";
+  const contentType = response.headers.get("content-type");
+
+  if (contentType?.includes("application/json")) {
+    const responseBody = await response.text();
+    let parsedBody: PdvApiResponseBody | PdvApiErrorBody | undefined;
+
+    try {
+      parsedBody = JSON.parse(responseBody);
+    } catch (error) {
+      logger.info(
```

Comment:
[nit] this is probably an `error` rather than `info? HMRC really shouldn't be sending anything other than json with the content type json...

Reply:
Alrighty can leave this for now then